core/domain/exp_services.py

```diff
@@ -723,12 +735,13 @@ def get_exploration_snapshots_metadata(exploration_id):
         in ascending order. There are exploration.version_number items in the
         returned list.
     """
-    exploration = get_exploration_by_id(exploration_id)
-    current_version = exploration.version
-    version_nums = range(1, current_version + 1)
+    if not max_version:
+        exploration = get_exploration_by_id(exploration_id)
+        max_version = exploration.version
```

Comment:
Don't you need to check this anyway (and take the min) even if a max_version is passed in? Otherwise you're going to run into problems if someone passes in 10^10.